packages/overlay/src/Overlay.ts

```diff
@@ -554,7 +556,10 @@ export class Overlay extends ComputedOverlayBase {
                 escapeDeactivates: false,
             });
 
-            if (this.type === 'modal' || this.type === 'page') {
+            if (
+                (this.type === 'modal' || this.type === 'page') &&
+                this.receivesFocus !== 'false'
```

Comment:
Move this condition to the initialFocus instead, because we want to have focus trap anyway